packages/e2e-tests/specs/editor/blocks/__snapshots__/buttons.test.js.snap

```diff
@@ -3,7 +3,7 @@
 exports[`Buttons can jump to the link editor using the keyboard shortcut 1`] = `
 "<!-- wp:buttons -->
 <div class=\\"wp-block-buttons\\"><!-- wp:button -->
-<div class=\\"wp-block-button\\"><a class=\\"wp-block-button__link\\">WordPress</a></div>
+<div class=\\"wp-block-button\\"><a class=\\"wp-block-button__link\\" href=\\"https://www.wordpress.org/\\" title=\\"\\">WordPress</a></div>
```

Comment:
Probably logical, not sure if it's explicitly related here. Also implemented in #19490.

Reply:
Noted also at https://github.com/WordPress/gutenberg/pull/19490#discussion_r366558464, but:

- An empty `title` is not useful here. In fact, depending on how it's interpreted, it could be actively harmful in stripping "WordPress" text as being the link's informative description.
- To that end: I'm not sure we really want to be messing with `title` at all. There are many resources which reinforce that `title` should _never_ be used on a link [[1]](https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute) [[2]](https://silktide.com/blog/2013/i-thought-title-text-improved-accessibility-i-was-wrong) [[3]](https://developer.paciellogroup.com/blog/2012/01/html5-accessibility-chops-title-attribute-use-and-abuse/) [[4]](https://www.24a11y.com/2017/the-trials-and-tribulations-of-the-title-attribute/) [[5]](https://a11yproject.com/posts/title-attributes/)
- Concerning the discussion at https://github.com/WordPress/gutenberg/pull/19462#discussion_r365204493 , we might thus want to not include `title` in the value object. Combine this with the fact that `opensInNewTab` is not always relevant for `value` (because it is only used when `settings` is the default), it really makes me wonder whether we should be representing a `value` as an object vs. as a string URL , and the settings values be managed separately (`onSettingsChange`?)

Reply:
> * To that end: I'm not sure we really want to be messing with `title` at all. There are many resources which reinforce that `title` should _never_ be used on a link [[1]](https://html.spec.whatwg.org/multipage/dom.html#the-title-attribute) [[2]](https://silktide.com/blog/2013/i-thought-title-text-improved-accessibility-i-was-wrong) [[3]](https://developer.paciellogroup.com/blog/2012/01/html5-accessibility-chops-title-attribute-use-and-abuse/) [[4]](https://www.24a11y.com/2017/the-trials-and-tribulations-of-the-title-attribute/) [[5]](https://a11yproject.com/posts/title-attributes/)

Also: https://core.trac.wordpress.org/ticket/24766

Reply:
in advanced situations in the Navigation block (and maybe others), we'll want to differentiate between relative links (post ids, category ids...) and absolute links using the same component. In that case, I believe an object is better to differentiate the different types of links.

Reply:
> in advanced situations in the Navigation block (and maybe others), we'll want to differentiate between relative links (post ids, category ids...) and absolute links using the same component. In that case, I believe an object is better to differentiate the different types of links.

Yeah, I can see this. I think the challenge here is in how we represent this, and if that's part of the `value` shape, how it's done in a way which is consistent and well-understood and also accommodates links which _aren't_ associated with known resources.

This ties in to the discussion at https://github.com/WordPress/gutenberg/pull/19462#discussion_r365204493 , but I don't think we had a complete understanding of the intended shape there, since I see [many references in code which seem to try to coerce everything into a post-like entity](https://github.com/WordPress/gutenberg/blob/b5c9419ed839a5f6735d65f35ab08d14ec933977/packages/block-editor/src/components/link-control/index.js#L96-L101).

It's taken me a while to get to this point, but I think I finally understand that this is inheriting from the [REST API search endpoint](https://github.com/WordPress/WordPress/blob/57fbd3c847c2499f6795e039d2be77c6ff5e323b/wp-includes/rest-api/endpoints/class-wp-rest-search-controller.php#L19-L42). I'm still trying to sort through some of the history there ([Trac#39965](https://core.trac.wordpress.org/ticket/39965)) because it seems to try to be generalized in a way which doesn't necessarily assume that these should be implied to always be _WordPress post entities_, which would certainly be helpful to support the "raw URL" usage.

With that in mind, maybe it's fine that the value contain all these fields. Merging them with the settings value gets a little more muddy, but I can see the value in convenience. I just wish we could define this in a way which is much more explicit (and consistent!) about what the consumer should be expecting to be available.

To the specific point about using `title` attributes, maybe it's not the problem that the search result includes a `title` property (I could see how it would be valuable to know the title of a resource associated with a search result), but more in how we use that property value to generate markup.